src/Miso/Canvas.hs

```diff
@@ -166,22 +166,25 @@ import           Miso.Html.Types
 import           Miso.Style (Color, renderColor)
 import           Miso.String (MisoString)
 -----------------------------------------------------------------------------
--- | Element for drawing on a <canvas>, includes a @Canvas@ DSL.
+-- | Element for drawing on a [\<canvas\>](https://developer.mozilla.org/en-US/docs/Web/HTML/Reference/Elements/canvas).
 -- This function abstracts over the context and interpret callback,
 -- including dimension ("2d" or "3d") canvas.
+--
+-- Unlike @Miso.HTML.Element.canvas@, which just creates an empty canvas element,
+-- this functions (TODO does what?)
 canvas
   :: forall action
    . [ Attribute action ]
-  -> JSM JSVal
+  -> JSM JSVal -- ^ TODO what does this argument represent?
   -> View action
 canvas attributes callback = node HTML "canvas" Nothing attrs []
   where
     attrs :: [ Attribute action ]
     attrs = flip (:) attributes $ Event $ \_ obj _ _ ->
       flip (FFI.set "draw") obj =<< callback
 -----------------------------------------------------------------------------
--- | Element for drawing on a <canvas>, includes a @Canvas@ DSL.
--- Specialized to "2d" canvas.
+-- | Element for drawing on a [\<canvas\>](https://developer.mozilla.org/en-US/docs/Web/HTML/Reference/Elements/canvas), includes a 'Canvas' DSL.
+-- Specialized to ["2d"](https://developer.mozilla.org/en-US/docs/Web/API/HTMLCanvasElement/getContext#2d) canvas.
```

Comment:
From the code it seems like "2d" is referring to one of the possible values of getContext method.
But I didn't find any docs on "3d".
I found https://stackoverflow.com/questions/9812237/can-you-use-canvas-getcontext3d-if-yes-how. I the accepted answer is right and "3d" doesn't exist, the docs shouldn't mention it